pkg/build/api/v1/types.go

```diff
@@ -402,6 +402,11 @@ type SourceBuildStrategy struct {
 
 	// ForcePull describes if the builder should pull the images from registry prior to building.
 	ForcePull bool `json:"forcePull,omitempty"`
+
+	// disableImplicitBuild prevents source-to-image from performing a Docker build operation
+	// when one is needed to execute ONBUILD statements in the builder image, or because
+	// the builder image does not include a prereq such as "sh" or "tar"
+	DisableImplicitBuild bool `json:"disableImplicitBuild,omitempty"`
```

Comment:
I don't like this being part of the build - it's not a responsibility of the author, it's a responsibility of the system to enforce this.